File: api-net/Controllers/ReportController.cs
using Liftonom.Api.Data;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace Liftonom.Api.Controllers;

[ApiController]
[Authorize]
[Route("api/v1/reports")]
public class ReportController(AppDbContext db) : ControllerBase
{
    private (DateTime Start, DateTime End) Range(DateTime? start, DateTime? end)
    {
        var s = (start ?? new DateTime(DateTime.UtcNow.Year, DateTime.UtcNow.Month, 1, 0, 0, 0, DateTimeKind.Utc)).Date;
        var e = (end ?? DateTime.UtcNow).Date.AddDays(1).AddTicks(-1);
        return (DateTime.SpecifyKind(s, DateTimeKind.Utc), DateTime.SpecifyKind(e, DateTimeKind.Utc));
    }

    [HttpGet("daily-summary")]
    public async Task<IActionResult> DailySummary([FromQuery] DateTime? date)
    {
        var d = (date ?? DateTime.UtcNow).Date;
        return Ok(new
        {
            date = DateOnly.FromDateTime(d).ToString("yyyy-MM-dd"),
            maintenance = await db.MaintenanceRecords.CountAsync(m => m.PlannedDate!.Value.Date == d),
            completed = await db.MaintenanceRecords.CountAsync(m => m.CompletedAt!.Value.Date == d),
            new_faults = await db.FaultReports.CountAsync(f => f.CreatedAt.Date == d),
            collections = await db.CashboxTransactions.Where(t => t.SourceType == "collection" && t.CreatedAt.Date == d).SumAsync(t => (decimal?)t.Amount) ?? 0,
            cash_in = await db.CashboxTransactions.Where(t => t.SourceType != "transfer" && t.Type == "in" && t.CreatedAt.Date == d).SumAsync(t => (decimal?)t.Amount) ?? 0,
            cash_out = await db.CashboxTransactions.Where(t => t.SourceType != "transfer" && t.Type == "out" && t.CreatedAt.Date == d).SumAsync(t => (decimal?)t.Amount) ?? 0,
        });
    }

    [HttpGet("collection-summary")]
    public async Task<IActionResult> CollectionSummary([FromQuery] DateTime? start, [FromQuery] DateTime? end)
    {
        var (s, e) = Range(start, end);
        var rows = await db.AccountTransactions
            .Where(t => t.SourceType == "collection" && t.CreatedAt >= s && t.CreatedAt <= e)
            .GroupBy(t => t.PaymentMethod)
            .Select(grp => new { payment_method = grp.Key, count = grp.Count(), total = grp.Sum(t => t.Amount) })
            .ToListAsync();
        return Ok(new { start = DateOnly.FromDateTime(s).ToString("yyyy-MM-dd"), end = DateOnly.FromDateTime(e).ToString("yyyy-MM-dd"),
            total = rows.Sum(r => r.total), by_method = rows });
    }

    [HttpGet("staff-performance")]
    public async Task<IActionResult> StaffPerformance([FromQuery] DateTime? start, [FromQuery] DateTime? end)
    {
        var (s, e) = Range(start, end);
        var users = await db.Users.Where(u => u.TenantId == db.CurrentTenantId && u.Role == "technician")
            .Select(u => new { u.Id, u.Name, u.Surname }).ToListAsync();

        var data = new List<object>();
        foreach (var u in users)
        {
            data.Add(new
            {
                user = $"{u.Name} {u.Surname}".Trim(),
                completed_maint = await db.MaintenanceRecords.CountAsync(m => m.Status == "completed" && m.CompletedAt >= s && m.CompletedAt <= e),
                resolved_faults = await db.FaultReports.CountAsync(f => f.AssignedUserId == u.Id && f.ResolvedAt >= s && f.ResolvedAt <= e),
            });
        }
        return Ok(new { start = DateOnly.FromDateTime(s).ToString("yyyy-MM-dd"), end = DateOnly.FromDateTime(e).ToString("yyyy-MM-dd"), data });
    }

    [HttpGet("inventory-movements")]
    public async Task<IActionResult> InventoryMovements([FromQuery] DateTime? start, [FromQuery] DateTime? end,
        [FromQuery(Name = "product_id")] long? productId)
    {
        var (s, e) = Range(start, end);
        var q = db.StockMovements.Where(m => m.CreatedAt >= s && m.CreatedAt <= e);
        if (productId is { } p) q = q.Where(m => m.ProductId == p);
        return Ok(new
        {
            start = DateOnly.FromDateTime(s).ToString("yyyy-MM-dd"),
            end = DateOnly.FromDateTime(e).ToString("yyyy-MM-dd"),
            @in = await q.Where(m => m.Type == "in").SumAsync(m => (decimal?)m.Quantity) ?? 0,
            @out = await q.Where(m => m.Type == "out").SumAsync(m => (decimal?)m.Quantity) ?? 0,
            items = await q.OrderByDescending(m => m.Id).Take(500).ToListAsync(),
        });
    }
}

File: api/app/Http/Controllers/Api/ReportController.php
<?php
// app/Http/Controllers/Api/ReportController.php

namespace App\Http\Controllers\Api;

use App\Http\Controllers\Controller;
use App\Models\CashboxTransaction;
use App\Models\FaultReport;
use App\Models\MaintenanceRecord;
use App\Models\StockMovement;
use App\Models\User;
use Illuminate\Http\Request;
use Illuminate\Support\Carbon;

class ReportController extends Controller
{
    /** Günlük özet: o günün bakım, tahsilat, arıza sayıları. */
    public function dailySummary(Request $request)
    {
        $date = $request->date('date') ?? today();

        return response()->json([
            'date'            => Carbon::parse($date)->toDateString(),
            'maintenance'     => MaintenanceRecord::whereDate('planned_date', $date)->count(),
            'completed'       => MaintenanceRecord::whereDate('completed_at', $date)->count(),
            'new_faults'      => FaultReport::whereDate('created_at', $date)->count(),
            'collections'     => (float) CashboxTransaction::where('source_type', 'collection')
                ->whereDate('created_at', $date)->sum('amount'),
            'cash_in'         => (float) CashboxTransaction::where('source_type', '!=', 'transfer')
                ->where('type', 'in')->whereDate('created_at', $date)->sum('amount'),
            'cash_out'        => (float) CashboxTransaction::where('source_type', '!=', 'transfer')
                ->where('type', 'out')->whereDate('created_at', $date)->sum('amount'),
        ]);
    }

    /** Tahsilat özeti: ödeme yöntemine göre gruplama. */
    public function collectionSummary(Request $request)
    {
        [$start, $end] = $this->range($request);

        $rows = \App\Models\AccountTransaction::query()
            ->where('source_type', 'collection')
            ->whereBetween('created_at', [$start, $end])
            ->selectRaw('payment_method, COUNT(*) as count, SUM(amount) as total')
            ->groupBy('payment_method')->get();

        return response()->json([
            'start' => $start->toDateString(),
            'end'   => $end->toDateString(),
            'total' => (float) $rows->sum('total'),
            'by_method' => $rows,
        ]);
    }

    /** Personel performansı: tamamlanan bakım / kapatılan arıza. */
    public function staffPerformance(Request $request)
    {
        [$start, $end] = $this->range($request);

        $users = User::where('tenant_id', $request->user()->tenant_id)
            ->where('role', 'technician')->get(['id', 'name', 'surname']);

        $data = $users->map(function ($u) use ($start, $end) {
            return [
                'user'              => trim($u->name . ' ' . $u->surname),
                'completed_maint'   => MaintenanceRecord::whereJsonContains('assigned_users', $u->id)
                    ->whereBetween('completed_at', [$start, $end])->where('status', 'completed')->count(),
                'resolved_faults'   => FaultReport::where('assigned_user_id', $u->id)
                    ->whereBetween('resolved_at', [$start, $end])->count(),
            ];
        });

        return response()->json(['start' => $start->toDateString(), 'end' => $end->toDateString(), 'data' => $data]);
    }

    /** Stok hareketleri raporu. */
    public function inventoryMovements(Request $request)
    {
        [$start, $end] = $this->range($request);

        $q = StockMovement::query()->whereBetween('created_at', [$start, $end]);
        if ($request->filled('product_id')) {
            $q->where('product_id', $request->integer('product_id'));
        }

        return response()->json([
            'start' => $start->toDateString(),
            'end'   => $end->toDateString(),
            'in'    => (float) (clone $q)->where('type', 'in')->sum('quantity'),
            'out'   => (float) (clone $q)->where('type', 'out')->sum('quantity'),
            'items' => $q->orderByDesc('id')->limit(500)->get(),
        ]);
    }

    private function range(Request $request): array
    {
        $start = $request->date('start') ?? now()->startOfMonth();
        $end   = $request->date('end') ?? now()->endOfMonth();

        return [Carbon::parse($start)->startOfDay(), Carbon::parse($end)->endOfDay()];
    }
}
